pkg/docker/util.go

```diff
@@ -187,46 +186,29 @@ func ParseImageReference(spec string) (ImageReference, error) {
 	var ref ImageReference
 
 	// TODO replace with docker version once docker/docker PR11109 is merged upstream
-	stream, tag, id := parseRepositoryTag(spec)
+	ref.Name, ref.Tag, ref.ID = parseRepositoryTag(spec)
 
-	repoParts := strings.Split(stream, "/")
-	switch len(repoParts) {
+	if len(ref.Name) == 0 {
+		return ImageReference{}, fmt.Errorf("the docker pull spec %q must not be empty", spec)
+	}
+
+	nameParts := strings.SplitN(ref.Name, "/", 3)
+	switch len(nameParts) {
 	case 2:
-		if strings.Contains(repoParts[0], ":") {
-			// registry/name
-			ref.Registry = repoParts[0]
-			ref.Namespace = "library"
-			ref.Name = repoParts[1]
-			ref.Tag = tag
-			ref.ID = id
-			return ref, nil
+		if strings.ContainsAny(nameParts[0], ":.") || nameParts[0] == "localhost" {
+			// registry/name, we suppose
+			ref.Registry = nameParts[0]
+			ref.Name = "library/" + nameParts[1]
```

Comment:
but don't we still need to add "library" in the case where "ref.Name" is something like "mysql"?